rest/polygon.go

```diff
@@ -0,0 +1,20 @@
+package polygon
+
+import (
+	"github.com/polygon-io/client-golang/rest/aggregates"
+	"github.com/polygon-io/client-golang/rest/client"
+)
+
+// todo: add comments for godoc
+
+type polygonClient struct {
+	Aggregates *aggregates.Client
+	// todo: Trades, Quotes, etc
+}
+
+func New(config client.HTTPBaseConfig) *polygonClient {
```

Comment:
I don't want to have to specify the domain or max retries unless they are different than the default. Most of the time the domain and max retries are fine with default values. API key is the only part which will surely be required to be known to create a client. Can we lower the required information to create the client to only what is required. With alternative methods for overriding defaults.